### PR TITLE
nodePackages.@uppy/companion: init at 3.1.1

### PR DESCRIPTION
**Pull request by netcrns**
###### Motivation for this change
Add companion server integration for Uppy.

https://github.com/transloadit/uppy/tree/main/packages/@uppy/companion

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by netcrns** [comment]
/marvin opt-in
/status needs_reviewer

I'm wondering, since this has not seen any review so far, should I just omit the generated changes in `pkgs/development/node-packages/node-packages.nix` to avoid conflicts?

**Comment by turion** [comment]
I'm not familiar enough with how node packages work. If you rebase onto current `master` and regenerate, will the diff in ` pkgs/development/node-packages/node-packages.nix` become smaller?

**Comment by netcrns** [comment]
> If you rebase onto current `master` and regenerate, will the diff in ` pkgs/development/node-packages/node-packages.nix` become smaller?

Aye, but the chances of new conflicts appearing in the very new future really makes the generation time of almost 2 hours quite bothersome.
/status awaiting_review

**Comment by winterqt** [comment]
@ofborg test thelounge

File: pkgs/development/node-packages/default.nix [hunk 1]
```diff
@@ -415,6 +415,10 @@ let
       buildInputs = [ pkgs.libusb1 ];
     };
 
+    uppy-companion = super."@uppy/companion".override {
+      name = "uppy-companion";
```
Comment (turion) [inline pkgs/development/node-packages/default.nix:419]: This is cumbersome. It sort of makes sense, but no other package does it.

Comment (turion) [reply pkgs/development/node-packages/default.nix:419]: I stand corrected. `bitwarden-cli` does it as well, and I wonder why not more packages that have `@` and `/` in the attribute name have it. LGTM then.

**Comment by turion** [comment]
> Aye, but the chances of new conflicts appearing in the very new future really makes the generation time of almost 2 hours quite bothersome.

True, that's always a nuisance when adding something to `nodePackages`. Can you still rebase and I'll try to merge quickly?

File: pkgs/development/node-packages/default.nix (hunk 1 above)
Comment (netcrns) [reply pkgs/development/node-packages/default.nix:419]: https://github.com/NixOS/nixpkgs/blob/6748e4d66376675407eaaaa2858786028242deeb/pkgs/development/node-packages/default.nix#L134-L137

**Comment by netcrns** [comment]
> Can you still rebase and I'll try to merge quickly?

I just did. ( ^_^)

**Pull request merged**